### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we fix warnings in oneDPL and in tests which were found in DPC++ compiler (win) + MS standard library :
- Warning G2904547A 'complex' is deprecated: warning STL4037: The effect of instantiating the template std::complex for any type other than float, double, or long double is unspecified....
- Warning G2FB9CE64 macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
- Warning G36382FF0 unused type alias '...' [-Wunused-local-typedef]
- Warning G3C3A837B lambda capture '...' is not used [-Wunused-lambda-capture]
- Warning G41AE70B8 unused variable '...' [-Wunused-variable] (partially fixed)
- Warning GE45D6FFB unused typedef '...' [-Wunused-local-typedef]
- Warning GF055929C unused function '...' [-Wunused-function]
- Linker error G4BA35A91 call to 'fpclassify' is ambiguous
```call to 'fpclassify' is ambiguous	out\build\x64-Debug C++17 dpcpp\github-oneapi	C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\corecrt_math.h	419```

### CODE REVIEW

File: test/xpu_api/numerics/numeric.ops/partial.sum/xpu_partial_sum.pass.cpp [hunk 1]
```diff
@@ -42,10 +42,11 @@ template <class InIter, class OutIter, class Test> void test() {
...
       auto out = buffer2.get_access<sycl_write>(cgh);
-      cgh.single_task<Test>([=]() {
-        OutIter r = oneapi::dpl::partial_sum(InIter(&in[0]), InIter(&in[0] + 5),
-                                             OutIter(&out[0]));
-      });
+            cgh.single_task<Test>(
+                [=]() {
+                    [[maybe_unused]] OutIter r =
+                        oneapi::dpl::partial_sum(InIter(&in[0]), InIter(&in[0] + 5), OutIter(&out[0]));
+                });
```
Comment (danhoeflinger) [inline test/xpu_api/numerics/numeric.ops/partial.sum/xpu_partial_sum.pass.cpp:49]: Seems like this is just formatting.

File: test/xpu_api/numerics/numeric.ops/partial.sum/xpu_partial_sum_op.pass.cpp [hunk 2]
```diff
@@ -43,11 +43,12 @@ template <class InIter, class OutIter, class Test> void test() {
...
-        OutIter r = oneapi::dpl::partial_sum(InIter(&in[0]), InIter(&in[0] + 5),
-                                             OutIter(&out[0]),
-                                             oneapi::dpl::minus<int>());
-      });
+            cgh.single_task<Test>(
+                [=]()
+                {
+                    [[maybe_unused]] OutIter r = oneapi::dpl::partial_sum(InIter(&in[0]), InIter(&in[0] + 5),
+                                                                          OutIter(&out[0]), oneapi::dpl::minus<int>());
+                });
```
Comment (danhoeflinger) [inline test/xpu_api/numerics/numeric.ops/partial.sum/xpu_partial_sum_op.pass.cpp:51]: Formatting only change.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 3]
```diff
@@ -248,7 +248,7 @@ test
 public:
     bool run(sycl::queue& queue)
     {
-        using result_type = typename Engine::scalar_type;
+        //using result_type = typename Engine::scalar_type;
```
Comment (danhoeflinger) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:251]: Delete instead of leave as comment?

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 4]
```diff
@@ -317,7 +317,7 @@ test<oneapi::dpl::ranlux24>
 public:
     bool run(sycl::queue& queue)
     {
-        using result_type = typename oneapi::dpl::ranlux24::scalar_type;
+        //using result_type = typename oneapi::dpl::ranlux24::scalar_type;
```
Comment (danhoeflinger) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:320]: Delete instead of comment?

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 3 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:251]: Fixed

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 4 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp]: Fixed

File: include/oneapi/dpl/pstl/algorithm_impl.h [hunk 5]
```diff
@@ -1450,15 +1450,18 @@ __remove_elements(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardI
...
             },
             ::std::plus<_DifferenceType>(),
-            [=](_DifferenceType __i, _DifferenceType __len, _DifferenceType __initial) {
+            [=](_DifferenceType __i, _DifferenceType __len, _DifferenceType __initial)
+            {
                 __internal::__brick_copy_by_mask(
                     __first + __i, __first + __i + __len, __result + __initial, __mask + __i,
-                    [](_ForwardIterator __x, _Tp* __z) {
+                    [](_ForwardIterator __x, _Tp* __z)
+                    {
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/algorithm_impl.h:1464]: No, not only:
from line 4115 removed ```__k``` from lambda capture list:
```[__first, __k, __n, __is_vector]``` -> ```[__first, __n, __is_vector]```
Other changes here - for CI clang format check.

File: test/xpu_api/numerics/complex.number/cmplx.over/real.pass.cpp [hunk 6]
```diff
@@ -12,6 +12,11 @@
 //   T
 //   real(const T& x);
 
+// Warning	G2904547A	'complex' is deprecated: warning STL4037:
+// The effect of instantiating the template std::complex for any type other than float, double, or long double is unspecified.
+// You can define _SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING to suppress this warning. [-Wdeprecated-declarations]
+#define _SILENCE_NONFLOATING_COMPLEX_DEPRECATION_WARNING
+
```
Comment (danhoeflinger) [inline test/xpu_api/numerics/complex.number/cmplx.over/real.pass.cpp:19]: Do we still want to test these types other than float, double, long double using `std::complex<>` if they are unspecified and compilation may fail according to the standard? 

If so, is there some way to check compiler support and enable testing only where it exists rather than just silencing the warning?

Comment (SergeyKopienko) [reply test/xpu_api/numerics/complex.number/cmplx.over/real.pass.cpp]: Fixed

File: test/xpu_api/numerics/complex.number/complex.member.ops/divide_equal_complex.pass.cpp [hunk 7]
```diff
@@ -49,7 +49,9 @@ ONEDPL_TEST_NUM_MAIN
 {
     test<float>();
     IF_DOUBLE_SUPPORT(test<double>())
+#if !_PSTL_ICC_TEST_COMPLEX_NON_FLOAT_COMPLEX_DEPRECATED
     IF_LONG_DOUBLE_SUPPORT(test<long double>())
+#endif // !_PSTL_ICC_TEST_COMPLEX_NON_FLOAT_COMPLEX_DEPRECATED
```
Comment (danhoeflinger) [inline test/xpu_api/numerics/complex.number/complex.member.ops/divide_equal_complex.pass.cpp:54]: Isn't `long double` one of the still supported types?  (seems to be done in 4 places)

Comment (SergeyKopienko) [reply test/xpu_api/numerics/complex.number/complex.member.ops/divide_equal_complex.pass.cpp]: Good point, fixed, thanks!

File: include/oneapi/dpl/pstl/algorithm_impl.h [hunk 8]
```diff
@@ -1456,15 +1456,15 @@ __remove_elements(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardI
...
-                __internal::__brick_copy_by_mask(
-                    __first + __i, __first + __i + __len, __result + __initial, __mask + __i,
-                    [](_ForwardIterator __x, _Tp* __z) {
-                        if constexpr (::std::is_trivial_v<_Tp>)
-                            *__z = ::std::move(*__x);
-                        else
-                            ::new (::std::addressof(*__z)) _Tp(::std::move(*__x));
-                    },
-                    __is_vector);
+                __internal::__brick_copy_by_mask(__first + __i, __first + __i + __len, __result + __initial,
```
Comment (rarutyun) [inline include/oneapi/dpl/pstl/algorithm_impl.h:1459]: What was changed in these lines?

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/algorithm_impl.h:1459]: Seems only clang formatting for CI

Comment (danhoeflinger) [reply include/oneapi/dpl/pstl/algorithm_impl.h:1459]: Quick plug for the clang format delta PR #715, which needs review and would have made the CI not suggest this change, away from the changed lines.

Comment (danhoeflinger) [reply include/oneapi/dpl/pstl/algorithm_impl.h:1459]: I believe one main result of our from our discussion on clang-format was that "it is not the boss".  

Therefore, I believe we have the choice if it is worth "touching" these lines in the git history for this formatting change.  In my opinion, I think its probably OK to revert this section, and merge with a failed clang-format github runner (for this specific section).

To me, they are equivalently readable, so it my vote would be to delay reformatting it until the code is actually "touched" by a real change.

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/algorithm_impl.h]: Fixed : restored old formatting.

File: test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp [hunk 9]
```diff
@@ -90,8 +90,8 @@ test(const ::std::size_t bits)
 
     const ::std::size_t max_n1 = 1000;
     const ::std::size_t max_n2 = (max_n1 * 10) / 8;
-    Sequence<T> in(max_n1, [&](::std::size_t) { return T(2 * HashBits(max_n1, bits - 1) ^ 1); });
-    Sequence<T> sub(max_n2, [&](::std::size_t) { return T(2 * HashBits(max_n1, bits - 1)); });
+    Sequence<T> in(max_n1, [bits](::std::size_t) { return T(2 * HashBits(max_n1, bits - 1) ^ 1); });
+    Sequence<T> sub(max_n2, [bits](::std::size_t) { return T(2 * HashBits(max_n1, bits - 1)); });
```
Comment (danhoeflinger) [inline test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp:94]: Why isn't an explicit or implicit capture of `max_n1` required?  Even though it is const, doesn't it still need to be captured in some way for use in the lambda?

Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp:94]: No, in this case it's not required. Otherwise we should have compile error.
In previous code we had the next warning here:
```
Warning G3C3A837B lambda capture 'max_n1' is not required to be captured for this use [-Wunused-lambda-capture]
```

Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp:94]: ...but in C++20 only...

Comment (danhoeflinger) [reply test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp:94]: Aha, I learned something new:  
https://en.cppreference.com/w/cpp/language/lambda

"A lambda expression can read the value of a variable without capturing it if the variable

has const non-volatile integral or enumeration type and has been initialized with a [constant expression](https://en.cppreference.com/w/cpp/language/constant_expression#Constant_expression), or
is constexpr and has no mutable members."

Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp]: To avoid compile error in C++17 I rolled back changes in this place.

**Review by danhoeflinger — APPROVED** [review]
LGTM

File: test/parallel_api/iterator/zip_iterator.pass.cpp [hunk 10]
```diff
@@ -614,7 +614,7 @@ DEFINE_TEST(test_merge)
         retrieve_data(host_keys, host_vals, host_res_merge);
 
         auto host_first1 = host_keys.get();
-        auto host_first2 = host_vals.get();
+        [[maybe_unused]] auto host_first2 = host_vals.get();
```
Comment (rarutyun) [inline test/parallel_api/iterator/zip_iterator.pass.cpp:617]: I mean it's not maybe_unused. It's unused. What is the purpose to keep this line?

File: test/parallel_api/numeric/numeric.ops/adjacent_difference.pass.cpp [hunk 11]
```diff
@@ -146,7 +146,7 @@ test(Pred pred)
 
     Sequence<T1> actual(max_len, [](::std::size_t i) { return T1(i); });
 
-    Sequence<T2> data(max_len, [value](::std::size_t i) { return i % 3 == 2 ? T2(i * i) : value; });
+    Sequence<T2> data(max_len, [&](::std::size_t i) { return i % 3 == 2 ? T2(i * i) : value; });
```
Comment (rarutyun) [inline test/parallel_api/numeric/numeric.ops/adjacent_difference.pass.cpp:149]: I believe the same comment as in test/parallel_api/algorithm/alg.nonmodifying/find_end.pass.cpp

File: test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp [hunk 12]
```diff
@@ -168,25 +171,29 @@ DEFINE_TEST(test_exclusive_scan_by_segment)
...
+            oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first, zero,
+                                                   [](KeyT first, KeyT second) { return first == second; });
         check_values(keys_first, val_res_first, zero, n);
 
         // call algorithm with addition operator
         initialize_data(keys_first, vals_first, val_res_first, n);
-        auto res4 = oneapi::dpl::exclusive_scan_by_segment(exec, keys_first, keys_last, vals_first, val_res_first,
-                                                           init, [](KeyT first, KeyT second) { return first == second; },
-                                                           [](ValT first, ValT second) { return first + second; });
+        [[maybe_unused]] auto res4 = oneapi::dpl::exclusive_scan_by_segment(
```
Comment (rarutyun) [inline test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp:193]: This is a slightly different, but again it's not maybe_unused. It's unused. Please investigate all similar cases but I believe we can just drop res4, res3 and all other variables like that in this file or in others (or probably it's good indication that we don't check the returned result anyhow, and probably we should)

File: test/support/utils_test_base.h [hunk 13]
```diff
@@ -319,10 +319,12 @@ struct test_base
 
 namespace
 {
+[[maybe_unused]]
```
Comment (rarutyun) [inline test/support/utils_test_base.h:322]: This fix looks super weird. The right thing to do it use this function properly.

Let's just make it 

template <typename TTestDataTransfer>
void retrieve_data(TTestDataTransfer& helper)
{
    helper.retrieve_data();
}

The same for the function below

File: test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp [hunk 14]
```diff
@@ -24,9 +24,11 @@
 #include "common_for_device_tests.h"
 #endif // TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
 
+#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
```
Comment (rarutyun) [inline test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp:27]: If looking more precisely all those variables can be moved under `#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS`

Comment (rarutyun) [reply test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp:27]: And I believe I have a lot of such places

**Comment by rarutyun** [comment]
I would like @aelizaro and or @paveldyakov to look at this patch (RNG related).

File: test/parallel_api/algorithm/alg.modifying.operations/unique_copy_equal.pass.cpp [hunk 15]
```diff
@@ -60,7 +60,7 @@ struct run_unique_copy
         ::std::fill_n(out_first, n, trash);
 
         // Run unique_copy
-        auto i = unique_copy(first, last, expected_first);
+        [[maybe_unused]] auto i = unique_copy(first, last, expected_first);
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.modifying.operations/unique_copy_equal.pass.cpp:63]: From my point of view this code is correct and we are unable to move this line into ```#else``` branch.

File: test/parallel_api/algorithm/alg.modifying.operations/alg.copy/copy_if.pass.cpp [hunk 16]
```diff
@@ -66,7 +66,7 @@ struct run_copy_if
         ::std::fill_n(out_first, n, trash);
 
         // Run copy_if
-        auto i = copy_if(first, last, expected_first, pred);
+        [[maybe_unused]] auto i = copy_if(first, last, expected_first, pred);
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/algorithm/alg.modifying.operations/alg.copy/copy_if.pass.cpp:69]: From my point of view this code is correct and we are unable to move this line into ```#else``` branch.

File: test/parallel_api/iterator/iterators.pass.cpp [hunk 17]
```diff
@@ -36,10 +36,10 @@ template <typename RandomIt>
...
     {
-        auto t1 = typename RandomIt::difference_type{};
-        auto t2 = typename RandomIt::value_type{};
-        auto t3 = typename RandomIt::pointer{};
-        typename RandomIt::reference ref = *it;
+        [[maybe_unused]] auto t1 = typename RandomIt::difference_type{};
+        [[maybe_unused]] auto t2 = typename RandomIt::value_type{};
+        [[maybe_unused]] auto t3 = typename RandomIt::pointer{};
+        [[maybe_unused]] typename RandomIt::reference ref = *it;
         (void) typename RandomIt::iterator_category{};
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/iterator/iterators.pass.cpp:43]: If we try to rewrite this last line like
```
[[maybe_unused]] typename RandomIt::iterator_category{};
```

we receive compile error:
```
test\parallel_api\iterator\iterators.pass.cpp(43,11): error G1C37D6F1: 'maybe_unused' attribute cannot be applied to a statement
         [[maybe_unused]] typename RandomIt::iterator_category{};
            ^                        ~~~~~~~~
```

File: test/parallel_api/iterator/zip_iterator.pass.cpp (hunk 10 above)
Comment (SergeyKopienko) [reply test/parallel_api/iterator/zip_iterator.pass.cpp]: Fixed.

File: test/parallel_api/numeric/numeric.ops/adjacent_difference.pass.cpp (hunk 11 above)
Comment (SergeyKopienko) [reply test/parallel_api/numeric/numeric.ops/adjacent_difference.pass.cpp]: Fixed.

File: test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp (hunk 12 above)
Comment (SergeyKopienko) [reply test/parallel_api/numeric/numeric.ops/exclusive_scan_by_segment.pass.cpp]: Reverted to initial state - will fix these warnings later.

File: test/support/utils_test_base.h (hunk 13 above)
Comment (SergeyKopienko) [reply test/support/utils_test_base.h:322]: Please explain.

File: test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp (hunk 14 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp:27]: Fixed.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 18]
```diff
@@ -202,7 +202,6 @@ test_vec<oneapi::dpl::ranlux24_vec<N>>
                         engine1.discard(offset);
                         typename oneapi::dpl::ranlux24_vec<N>::result_type res0;
                         oneapi::dpl::ranlux24_vec<N> engine(engine1);
-                        auto eng = engine.base();
```
Comment (paveldyakov) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:205]: This line shouldn't be removed as it is a simple check of API. Like that .base() is callable from this engine.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 19]
```diff
@@ -342,7 +337,6 @@ test<oneapi::dpl::ranlux24>
                         engine1.discard(offset);
                         typename oneapi::dpl::ranlux24::result_type res0;
                         oneapi::dpl::ranlux24 engine(engine1);
-                        auto eng = engine.base();
```
Comment (paveldyakov) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:345]: This line shouldn't be removed as it is a simple check of API. Like that .base() is callable from this engine.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 20]
```diff
@@ -248,8 +247,6 @@ test
 public:
     bool run(sycl::queue& queue)
     {
-        using result_type = typename Engine::scalar_type;
```
Comment (paveldyakov) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:251]: The same for that - it is a simple check of API.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp [hunk 21]
```diff
@@ -317,8 +314,6 @@ test<oneapi::dpl::ranlux24>
 public:
     bool run(sycl::queue& queue)
     {
-        using result_type = typename oneapi::dpl::ranlux24::scalar_type;
```
Comment (paveldyakov) [inline test/xpu_api/random/interface_tests/engines_methods.pass.cpp:320]: The same for that - it is a simple check of API.

Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:320]: Fixed

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 20 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:251]: Fixed

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 19 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:345]: Fixed.

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 18 above)
Comment (SergeyKopienko) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:205]: Fixed.

**Review by paveldyakov — DISMISSED** [review]

**Review by paveldyakov — DISMISSED** [review]

File: test/parallel_api/iterator/iterators.pass.cpp (hunk 17 above)
[the remark this answers is not in the document]
Comment (rarutyun) [reply test/parallel_api/iterator/iterators.pass.cpp:43]: I've never proposed it to be like `[[maybe_unused]] typename RandomIt::iterator_category{};`

I would rather proposed it to be like `[[maybe_unused]] auto it_category = typename RandomIt::iterator_category{};`

File: test/parallel_api/algorithm/alg.modifying.operations/replace.pass.cpp [hunk 22]
```diff
@@ -122,7 +122,7 @@ test(Pred pred)
     Sequence<T2> expected(max_len);
     Sequence<T2> actual(max_len);
 
-    Sequence<T2> data(max_len, [&value](::std::size_t i) {
```
Comment (rarutyun) [inline test/parallel_api/algorithm/alg.modifying.operations/replace.pass.cpp:125]: Here we made the capturing by reference (unlike other places where you changed it to `[=]`) so in that line on code I would prefer `[&]`.

File: test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp (hunk 14 above)
Comment (rarutyun) [reply test/xpu_api/random/device_tests/weibull_distribution_device_test.pass.cpp:27]: That's not what I was asked for. I didn't ask to wrap them with macros. I asked to move them under already existing macros. Basically in `main` function under  
`#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS`

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 18 above)
Comment (rarutyun) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:205]: @paveldyakov , then I believe you should check somehow this `eng` variable.

Is not is better to leave it as is (without `[[maybe_unused]]`) to indicate there is something you should fix?

File: test/support/utils_test_base.h [hunk 23]
```diff
@@ -319,14 +319,20 @@ struct test_base
 
 namespace
```
Comment (rarutyun) [inline test/support/utils_test_base.h:320]: you can remove this unnamed `namespace` because it stops being helpful here

File: test/support/test_config.h [hunk 24]
```diff
@@ -76,7 +76,9 @@
 #endif
 
 // Enable test when the TBB backend is available
-#define TEST_TBB_BACKEND_PRESENT (!defined(ONEDPL_USE_TBB_BACKEND) || ONEDPL_USE_TBB_BACKEND != 0)
+#if !defined(ONEDPL_USE_TBB_BACKEND) || ONEDPL_USE_TBB_BACKEND
```
Comment (rarutyun) [inline test/support/test_config.h:79]: Don't understand this one.

File: test/support/test_config.h [hunk 25]
```diff
@@ -110,6 +112,9 @@
 #define _PSTL_ICC_TEST_COMPLEX_POW_SCALAR_COMPLEX_PASS_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
 #define _PSTL_ICC_TEST_COMPLEX_NORM_MINUS_INF_NAN_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
 #define _PSTL_ICC_TEST_COMPLEX_POLAR_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
+#define _PSTL_ICC_TEST_COMPLEX_NON_FLOAT_COMPLEX_DEPRECATED (_MSC_VER >= 1930)
```
Comment (rarutyun) [inline test/support/test_config.h:115]: Don't understand why those macros are part of this PR

File: test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp [hunk 26]
```diff
@@ -21,8 +21,9 @@
 #include <cassert>
 #include <iostream>
 
-constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+#if !defined(_WIN32)
```
Comment (rarutyun) [inline test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp:24]: It's completely unclear to me what we are doing here

File: test/xpu_api/random/device_tests/bernoulli_distribution_device_test.pass.cpp [hunk 27]
```diff
@@ -24,9 +24,11 @@
 #include "common_for_device_tests.h"
 #endif // TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
 
+#if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
```
[the remark this answers is not in the document]
Comment (paveldyakov) [reply test/xpu_api/random/device_tests/bernoulli_distribution_device_test.pass.cpp:27]: Such approach just let us to find and modify the variables in test more simpler. Also it is the test, why it is important to have the variables not global?

File: test/xpu_api/random/interface_tests/engines_methods.pass.cpp (hunk 18 above)
Comment (paveldyakov) [reply test/xpu_api/random/interface_tests/engines_methods.pass.cpp:205]: @rarutyun, the API call is also fine. If API doesn't exist we will have the error on compilation stage.

File: test/support/test_config.h (hunk 24 above)
Comment (SergeyKopienko) [reply test/support/test_config.h:79]: Because otherwise we had the next warning when use this macros in code:
```
test/support/test_config.h - Warning G2FB9CE64 macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
```

File: test/parallel_api/iterator/iterators.pass.cpp (hunk 17 above)
[the remark this answers is not in the document]
Comment (akukanov) [reply test/parallel_api/iterator/iterators.pass.cpp:43]: I agree with Ruslan here, `[[maybe_unused]] auto t4 = typename RandomIt::iterator_category{};` would be consistent with the checks above.

File: test/xpu_api/random/default_engine_params.h [hunk 28]
```diff
@@ -0,0 +1,27 @@
...
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------===//
+
+#ifndef _DEFAULT_ENGINE_PARAMS_H
+#define _DEFAULT_ENGINE_PARAMS_H
+
+// Default engine params
+namespace DefaultEngineParams
```
Comment (akukanov) [inline test/xpu_api/random/default_engine_params.h:19]: As we can see, this resulted in massive changes and unnecessarily increased verbosity in all RNG tests.

I suggest to spin it off into a separate PR, or maybe even ask @paveldyakov and his team to propose a solution that they think is most appropriate (as they are the maintainers of these tests).

File: test/parallel_api/iterator/iterators.pass.cpp (hunk 17 above)
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/parallel_api/iterator/iterators.pass.cpp:43]: Fixed

File: test/support/test_config.h [hunk 29]
```diff
@@ -110,6 +112,9 @@
 #define _PSTL_ICC_TEST_COMPLEX_POW_SCALAR_COMPLEX_PASS_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
 #define _PSTL_ICC_TEST_COMPLEX_NORM_MINUS_INF_NAN_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
 #define _PSTL_ICC_TEST_COMPLEX_POLAR_BROKEN_TEST_EDGES __INTEL_LLVM_COMPILER
+#define _PSTL_ICC_TEST_COMPLEX_NON_FLOAT_COMPLEX_DEPRECATED (_MSVC_STL_VERSION >= 143)
```
[the remark this answers is not in the document]
Comment (SergeyKopienko) [reply test/support/test_config.h:115]: Agree. I like variant to have ```_PSTL_TEST_COMPLEX_NON_FLOAT_AVAILABLE``` macro.
Fixed.

File: test/support/utils_test_base.h (hunk 23 above)
Comment (SergeyKopienko) [reply test/support/utils_test_base.h]: Fixed

**Review by akukanov — APPROVED** [review]
